Makefile

```diff
@@ -42,6 +43,8 @@ $(WEAVER_EXE) $(WEAVEDNS_EXE): common/*.go
 
 $(WEAVER_EXE): router/*.go weaver/main.go
 $(WEAVEDNS_EXE): nameserver/*.go weavedns/main.go
+$(SIGPROXY_EXE): sigproxy/main.go
+	go build -o $@ ./$(shell dirname $@)
```

Comment:
Have you checked that this isn't dragging in any dynamically linked libs? I can't remember the precise reasons for the ldflags magic in the main build stanza.

Also, I suggest you add a comment explaining why this isn't using the main build stanza.